Userland/Libraries/LibJS/Runtime/ArrayBufferPrototype.cpp

```diff
@@ -66,12 +66,48 @@ static ArrayBuffer* array_buffer_object_from(VM& vm, GlobalObject& global_object
     return static_cast<ArrayBuffer*>(&this_object);
 }
 
+// 25.1.5.3 ArrayBuffer.prototype.slice, https://tc39.es/ecma262/#sec-arraybuffer.prototype.slice
 JS_DEFINE_NATIVE_FUNCTION(ArrayBufferPrototype::slice)
 {
+    const auto start = vm.argument(0);
+    const auto end = vm.argument(1);
+
     auto array_buffer_object = array_buffer_object_from(vm, global_object);
     if (!array_buffer_object)
         return {};
-    TODO();
+
+    // FIXME: Check for shared buffer
+    // FIXME: Check for detached buffer
+
+    const auto len = array_buffer_object->byte_length();
+
+    const auto relative_start = start.is_negative_infinity()
+        ? 0
+        : start.to_integer_or_infinity(global_object);
+    const auto first = relative_start < 0
+        ? AK::max(len + relative_start, double(0))
+        : AK::min(relative_start, double(len));
+
+    const auto relative_end = end.is_undefined()
+        ? len
+        : end.to_integer_or_infinity(global_object);
+
+    double final;
+    if (end.is_negative_infinity())
+        final = 0;
+    else if (relative_end < 0)
+        final = AK::max(len + relative_end, double(0));
+    else
+        final = AK::min(relative_end, double(len));
+
+    const auto new_len = AK::max(final - first, double(0));
+
+    auto buffer = ArrayBuffer::create(global_object, new_len);
```

Comment:
This is a bit more sophisticated in the spec to work with `ArrayBuffer` subclasses:

```
15. Let ctor be ? SpeciesConstructor(O, %ArrayBuffer%).
16. Let new be ? Construct(ctor, « 𝔽(newLen) »).
17. Perform ? RequireInternalSlot(new, [[ArrayBufferData]]).
(18. If IsSharedArrayBuffer(new) is true, throw a TypeError exception.)
(19. If IsDetachedBuffer(new) is true, throw a TypeError exception.)
20. If SameValue(new, O) is true, throw a TypeError exception.
21. If new.[[ArrayBufferByteLength]] < newLen, throw a TypeError exception.
(22. NOTE: Side-effects of the above steps may have detached O.)
(23. If IsDetachedBuffer(O) is true, throw a TypeError exception.)
```

Happy to do this for now, but please add a FIXME for our future selves if so :)